### PR TITLE
Add typing support

### LINKED ISSUE #253: Typing data for feedparser
Currently trying to run mypy against a project with feedparser gets

`error: Skipping analyzing 'feedparser': found module but no type hints or library stubs`

### PR DESCRIPTION
**Pull request by palfrey**
Fixes #253

### CODE REVIEW

File: feedparser/encodings.py [hunk 1]
```diff
@@ -32,11 +32,10 @@
 
 try:
     try:
-        import cchardet as chardet
+        import chardet as chardet
     except ImportError:
         import chardet
```
Comment (kurtmckee) [inline feedparser/encodings.py:37]: It looks like this loses the cchardet import and tries to import chardet twice. Would you double-check this behavior?

File: feedparser/sgml.py [hunk 2]
```diff
@@ -27,7 +27,8 @@
 
 import re
 
-import sgmllib
+# FIXME: replace with something else, as sgmllib3k is not maintained. See https://github.com/kurtmckee/feedparser/issues/279
```
Comment (kurtmckee) [inline feedparser/sgml.py:30]: This is not needed for this patch.

**Review by kurtmckee — CHANGES_REQUESTED** [review]
Great work, Tom! I see a couple of things to improve:

1. Please use scriv to add a changelog entry.
2. The unit tests for PyPy3 are failing. Please review to see if this can be improved.
3. Please double-check the cchardet import, which seems to result in incorrect behavior now.
4. No need to leave a comment about sgmllib3k, which is not related to this patch.

Thanks so much for submitting this!

File: feedparser/encodings.py (hunk 1 above)
Comment (palfrey) [reply feedparser/encodings.py]: I didn't actually realise this was a different library as I thought it was a spelling mistake I think! Out of curiosity: why are neither listed as dependencies of this library?

File: feedparser/sgml.py (hunk 2 above)
Comment (palfrey) [reply feedparser/sgml.py]: I wanted to flag it further as it's lack of updating means there's no mypy work for it at all coming. I've removed it though.

**Comment by palfrey** [comment]
> Great work, Tom! I see a couple of things to improve:
>     1. Please use scriv to add a changelog entry.

Done

>     2. The unit tests for PyPy3 are failing. Please review to see if this can be improved.

This appears to be an issue with typed-ast which is depended on by Sphinx. I've disabled Sphinx install on pypy for the moment.

>     3. Please double-check the cchardet import, which seems to result in incorrect behavior now.
> 
>     4. No need to leave a comment about sgmllib3k, which is not related to this patch.

Fixed

File: feedparser/encodings.py (hunk 1 above)
Comment (kurtmckee) [reply feedparser/encodings.py]: It's optional -- developers can install these libraries for additional flexibility but they're not required.

**Review by kurtmckee — APPROVED** [review]

**Comment by kurtmckee** [comment]
Thanks for your work on this, Tom!

**Review by yvettep321 — COMMENTED** [review]
thank you

**Comment by Coinhexa** [comment]
may I know when are you guys planning to push this into the master branch?

**Comment by NLaundry** [comment]
> may I know when are you guys planning to push this into the master branch?

Would love to know this too.

**Comment by kurtmckee** [comment]
The code in the `develop` branch requires work to prep for a v7.0.0 release. I will work to get this into shape, but it will take time, and I don't have an ETA.

**Comment by anon-dev-gh** [comment]
> The code in the `develop` branch requires work to prep for a v7.0.0 release. I will work to get this into shape, but it will take time, and I don't have an ETA.

Is there any way in which I can help?

**Pull request merged**